x-pack/legacy/plugins/siem/public/containers/case/configure/use_configure.tsx

```diff
@@ -4,47 +4,203 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { useState, useEffect, useCallback } from 'react';
+import { useEffect, useCallback, useReducer } from 'react';
 import { getCaseConfigure, patchCaseConfigure, postCaseConfigure } from './api';
 
 import { useStateToaster, errorToToaster, displaySuccessToast } from '../../../components/toasters';
 import * as i18n from './translations';
-import { ClosureType } from './types';
-import { CurrentConfiguration } from '../../../pages/case/components/configure_cases/reducer';
+import { CasesConfigurationMapping, ClosureType } from './types';
 
-export interface PersistCaseConfigure {
+interface Connector {
   connectorId: string;
   connectorName: string;
+}
+export interface ConnectorConfiguration extends Connector {
```

Comment:
What do you think about changing the type to:

```
export interface ConnectorConfiguration extends Connector {
  connector: Connector;
  closureType: ClosureType;
}
```

In the future, the connector's `actionTypeId` will be needed so it will be nice to have an object that represents the connector.

Reply:
yes but do you mean to have `extends Connector` still?

Reply:
No my mistake :)